tests/test_utils.py

```diff
@@ -64,4 +64,62 @@ def test_run_command(caplog):
         and_return(subprocess.CompletedProcess(cmd, 0, "stdout", "stderr"))
     res = run_command(cmd, log_level=log_level, additional_env=additional_env, combine_out_err=combine_out_err)
     assert secret not in caplog.text
-    assert REDACTED_OUT_SECRET in caplog.text
\ No newline at end of file
+    assert REDACTED_OUT_SECRET in caplog.text
+
+
+def test_change_cwd():
+    present_dir = os.getcwd()
+    with tempfile.TemporaryDirectory() as target_dir:
+        with change_cwd(target_dir):
+            assert os.getcwd() == os.path.realpath(target_dir)
+        assert os.getcwd() == present_dir
+
+
+def test_get_current_commit(caplog):
+    current_commit = subprocess.run(["git", "rev-parse", "--short", "HEAD"],text=True, capture_output=True)
+    assert current_commit.stdout.strip() == get_current_commit(".")
+    with caplog.at_level(logging.WARNING):
+        with tempfile.TemporaryDirectory() as target_dir:
+            get_current_commit(target_dir)
+            for record in caplog.records:
+                assert "Failed getting current git commit" in record
+
+
+def test_validate_duplicates():
+    with pytest.raises(ValueError) as err:
+        validate_duplicates(["a", "b"], ["b", "c"])
+        assert "Duplicate field" in str(err.value)
+
+
+
+def test_fmt_cmd_out_for_log():
+    fake_CalledProcessError = flexmock.flexmock(returncode=1, stdout="stdout", stderr="stderr")
+    result = fmt_cmd_out_for_log(fake_CalledProcessError, combine_out_err=True)
+    assert result == 'RETCODE: 1\nOUTPUT:\nstdout'
```

Comment:
nit: Can you add a newline after each assert to show its visually different tests ?